tests/tcl/tests/unit/scripting.tcl

```diff
@@ -263,9 +263,10 @@ start_server {tags {"scripting"}} {
                 assert(re.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x)
                 -- so, the final x.x is at the depth limit and was assigned nil
                 assert(re.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x.x == nil)
-                return {h, re.x.x.x.x.x.x.x.x.y == re.y, re.y == 5}
+                assert(h == "82a17881a17882a17881a17882a17881a17882a17881a17882a17881a17882a17881a17882a17881a17882a17881a178c0a17905a17905a17905a17905a17905a17905a17905a17905" or h == "82a17905a17881a17882a17905a17881a17882a17905a17881a17882a17905a17881a17882a17905a17881a17882a17905a17881a17882a17905a17881a17882a17905a17881a178c0")
```

Comment:
Why there are two different value of `h`?

Reply:
I had fix this test. The case is from here https://github.com/antirez/lua-cmsgpack/blob/master/test.lua#L393

Reply:
I got it. It is due to the element order of a table is unstable. This reminds me of Map in golang.

Reply:
Yes， It behaves differently in different compilers

Reply:
> Yes， It behaves differently in different compilers

I still have deep doubts about this, because I think we should try to avoid semantics behaving differently under different compilers.